validations/src/validations.rs

```diff
@@ -1753,10 +1753,22 @@ pub fn validate_block(
         // with the genesis_block_hash
         validate_genesis_block(block, consensus_constants.genesis_hash).map_err(Into::into)
     } else {
-        let total_identities = u32::try_from(rep_eng.ars().active_identities_number())?;
+        let (total_identities, replication_factor) =
+            if after_second_hard_fork(current_epoch, get_environment()) {
```

Comment:
Not sure if `current_epoch == block_epoch` here

```suggestion
            if after_second_hard_fork(block_epoch, get_environment()) {
```